pkg/sources/docker/docker.go

```diff
@@ -181,6 +206,86 @@ func (s *Source) processImage(ctx context.Context, image string) (imageInfo, err
 	return imgInfo, nil
 }
 
+// getHistoryEntries collates an image's configuration history together with the
+// corresponding layer digests for any non-empty layers.
+func getHistoryEntries(ctx context.Context, imgInfo imageInfo) ([]historyEntryInfo, error) {
+	config, err := imgInfo.image.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+
+	layers, err := imgInfo.image.Layers()
+	if err != nil {
+		return nil, err
+	}
+
+	history := config.History
+	entries := make([]historyEntryInfo, len(history))
+
+	layerIndex := 0
+	for historyIndex, entry := range history {
+		e := historyEntryInfo{
+			base:  imgInfo.base,
+			tag:   imgInfo.tag,
+			entry: entry,
+			index: historyIndex,
+		}
+
+		// Associate with a layer if possible -- failing to do this will not affect
+		// the scan, just remove some traceability.
+		if !entry.EmptyLayer {
```

Comment:
I absolutely agree! The only reason I didn't do this is to avoid repeating the assignment into the array and incrementing the layer index.

Keeping those separate means that there is less chance of introducing a bug (by omitting the increment or assignment) if there is additional logic added into this segment.

Regardless, however, I'd prefer to align with the preferences of those maintaining the code-base as it's really you that I'm trying to optimise for. 

With that in mind, given the above reasoning, what would you prefer to maintain? I'll happily align with your suggestion!

Reply:
That’s a compelling reason. Let’s leave it as is. Thanks for the explanation.

Reply:
And I really appreciate your thoughtful and clear commentary. 💪